### PR TITLE
Add GitHub Gist extension

### PR DESCRIPTION
**Pull request by koinzhang**
## Description

**Getting Started**

This extension brings GitHub Gist support to Raycast through the personal access tokens. To get started, first:

- Login to your  GitHub instance (Eg. [https://github.com/](https://github.com))
- Click on your avatar image in the right upper corner
- From the dropdown menu, click on `Settings`
- On the left-hand side, click on `Developer settings`.
- On the left-hand side, click on `Personal access tokens`.
- On this page, we'll create a new access token. Click on `Generate new token` in the upper-right.
- Leave a `note` for your token (Eg. `Raycast`). This will help you identify it in the future.
- You'll need to check the following boxes to ensure this extension can perform properly:
  - `repo`
  - `gist`
  - `user`
- Click `Generate token` and save this value somewhere. **You'll only be able to see this once**.

> Future updates to this extension will offer more functionality that may require additional scopes be defined in this token.

**Create Gist**

Automatically get the selected text or clipboard and generate a gistfile, you can also add additional gistfiles and finally create a gist.
Once created successfully, you can choose to copy the link or open it in your browser.

**Search Gist**

Search and display gists from GitHub.

- Gist Tag

  - My Gists

  - My Starred

  - All Public Gists

- Action

  - Copy to Clipboard

  - Paste to Active App

  - Copy Gist Link

  - Open in Browser
  - Star Gist *(Only in MyGists)*
  - Delete Gist *(Only in MyGists)*

  - Unstar Gist *(Only in My Starred)*

  - Star Gist *(Only in All Public Gists)*

  - Clone Gist

## Screencast

https://user-images.githubusercontent.com/36128970/159161962-70adae29-29c5-4026-82d0-19f23b8dcc22.mp4

https://user-images.githubusercontent.com/36128970/159211152-65b06683-56a3-4db4-842f-3acb05c6ce62.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

**Comment by andreaselia** [comment]
Great work! I feel like it would be nice to have a loading state when switching between My/Starred/All Gists.

As a feature suggestion, would it be worth adding the ability to create a Gist from whatever is copied from your clipboard? This would create a Gist from whatever is in your clipboard, and return the URL to said Gist to your clipboard.

**Comment by koinzhang** [comment]
hi @andreaselia, Thank you for the review.
Changes in this commit :
1. Add Loading state for time-consuming network operations to improve user experience,
2. Add the command to create gist,
Please continue to review.

File: extensions/github-gist/package.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          "required": false,
+          "title": "Primary action",
+          "default": "copy",
+          "data": [
+            {
+              "title": "Copy to Clipboard",
+              "value": "copy"
+            },
+            {
+              "title": "Paste to Active APP",
```
Comment (fe9lix) [inline extensions/github-gist/package.json:60]:
```suggestion
              "title": "Paste to Active App",
```

File: extensions/github-gist/src/create-gist.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                      primaryAction: {
+                        title: "Copy Gist Link",
+                        onAction: (toast) => {
+                          Clipboard.copy(String(response.data.html_url));
+                          toast.title = "Link is copied to Clipboard";
+                          toast.message = "";
+                        },
+                      },
+                      secondaryAction: {
+                        title: "Open in browser",
```
Comment (fe9lix) [inline extensions/github-gist/src/create-gist.tsx:91]:
```suggestion
                        title: "Open in Browser",
```

File: extensions/github-gist/src/create-gist.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                  if (isEdit) {
+                    response = await updateGist(gist.gist_id, description, files);
+                  } else {
+                    response = await createGist(description, isPublic, files);
+                  }
+                  if (response.status === 201 || response.status === 200) {
+                    const options: Toast.Options = {
+                      style: Toast.Style.Success,
+                      title: isEdit ? "Update Gist Success" : "Create Gist Success",
+                      message: "Click to copy gist link",
```
Comment (fe9lix) [inline extensions/github-gist/src/create-gist.tsx:81]:
```suggestion
                      message: "Click to copy Gist link",
```

**Review by fe9lix — CHANGES_REQUESTED** [review]

**Comment by fe9lix** [comment]
@koinzhang Cheers, just fixed some typos...

**Pull request merged**